src/App.tsx

```diff
@@ -15,36 +16,111 @@ export const goodsFromServer = [
+
+function getPreparedGoods(
+  goods: string[],
+  { sortField }: { sortField: SortType },
+): string[] {
+  const preparedGoods = [...goods];
+
+  if (sortField) {
+    switch (sortField) {
+      case SORT_FIELD_ALPHABET:
+        preparedGoods.sort((a, b) => a.localeCompare(b));
+        break;
+      case SORT_FIELD_LENGTH:
+        preparedGoods.sort((a, b) => a.length - b.length);
+        break;
+      default:
+        break;
+    }
+  }
+
+  return preparedGoods;
+}
+
 export const App: React.FC = () => {
+  const [sortField, setSortField] = useState<SortType | ''>('');
+
+  let visibleGoods = sortField
+    ? getPreparedGoods(goodsFromServer, { sortField })
+    : [...goodsFromServer];
+
+  const [reversed, setReversed] = useState(false);
+
+  if (reversed) {
+    visibleGoods = visibleGoods.reverse();
+  }
+
+  const buttonClassAplhabet = classNames('button', 'is-info', {
+    'is-light': sortField !== SORT_FIELD_ALPHABET,
+  });
+
+  const buttonClassLength = classNames('button', 'is-success', {
+    'is-light': sortField !== SORT_FIELD_LENGTH,
+  });
+
+  const buttonClassReverse = classNames('button', 'is-warning', {
+    'is-light': !reversed,
+  });
+
   return (
     <div className="section content">
       <div className="buttons">
-        <button type="button" className="button is-info is-light">
+        <button
+          type="button"
+          className={buttonClassAplhabet}
+          onClick={() => setSortField(SORT_FIELD_ALPHABET)}
+        >
           Sort alphabetically
         </button>
 
-        <button type="button" className="button is-success is-light">
+        <button
+          type="button"
+          className={buttonClassLength}
+          onClick={() => setSortField(SORT_FIELD_LENGTH)}
+        >
           Sort by length
         </button>
 
-        <button type="button" className="button is-warning is-light">
+        <button
+          type="button"
+          className={buttonClassReverse}
+          onClick={() => setReversed(!reversed)}
+        >
           Reverse
         </button>
 
-        <button type="button" className="button is-danger is-light">
-          Reset
-        </button>
+        {(sortField || reversed) && (
+          <button
+            type="button"
+            className="button is-danger is-light"
+            onClick={() => {
+              setSortField('');
+              setReversed(false);
+            }}
+          >
+            Reset
+          </button>
+        )}
       </div>
 
       <ul>
-        <ul>
-          <li data-cy="Good">Dumplings</li>
-          <li data-cy="Good">Carrot</li>
-          <li data-cy="Good">Eggs</li>
-          <li data-cy="Good">Ice cream</li>
-          <li data-cy="Good">Apple</li>
-          <li data-cy="Good">...</li>
-        </ul>
+        {visibleGoods.map(goods => {
+          return (
+            <li key={goods} data-cy="Good">
+              {goods}
+            </li>
+          );
+        })}
```

Comment:
According to the checklist, when rendering a list, you should add a `key` prop that is unique for each item. Here, you're using `goods` as a key, which might not be unique if the list contains duplicate items. Consider using the index of the map function or a unique identifier.